bitnami/harbor/templates/_helpers.tpl

```diff
@@ -0,0 +1,518 @@
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "harbor.fullname" -}}
+{{- $name := default "harbor" .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/* Helm required labels */}}
+{{- define "harbor.labels" -}}
+app.kubernetes.io/name: {{ template "harbor.name" . }}
+helm.sh/chart: {{ template "harbor.chart" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end -}}
+
+{{/* matchLabels */}}
+{{- define "harbor.matchLabels" -}}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/name: {{ template "harbor.name" . }}
+{{- end -}}
+
+{{- define "harbor.autoGenCert" -}}
+  {{- if and .Values.service.tls.enabled (not .Values.service.tls.secretName) -}}
+    {{- printf "true" -}}
+  {{- else -}}
+    {{- printf "false" -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "harbor.autoGenCertForIngress" -}}
+  {{- if and (eq (include "harbor.autoGenCert" .) "true") (eq .Values.service.type "Ingress") -}}
+    {{- printf "true" -}}
+  {{- else -}}
+    {{- printf "false" -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "harbor.autoGenCertForNginx" -}}
+  {{- if and (eq (include "harbor.autoGenCert" .) "true") (ne .Values.service.type "Ingress") -}}
+    {{- printf "true" -}}
+  {{- else -}}
+    {{- printf "false" -}}
+  {{- end -}}
+{{- end -}}
+
+Create a default fully qualified postgresql name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "harbor.postgresql.fullname" -}}
+{{- $name := default "postgresql" .Values.postgresql.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "harbor.database.host" -}}
+  {{- if eq .Values.postgresql.enabled true -}}
+    {{- template "harbor.postgresql.fullname" . }}
+  {{- else -}}
+    {{- .Values.externalDatabase.host -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "harbor.database.port" -}}
+  {{- if eq .Values.postgresql.enabled true -}}
+    {{- printf "%s" "5432" -}}
+  {{- else -}}
+    {{- .Values.externalDatabase.port -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "harbor.database.username" -}}
+  {{- if eq .Values.postgresql.enabled true -}}
+    {{- printf "%s" "postgres" -}}
+  {{- else -}}
+    {{- .Values.externalDatabase.user -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "harbor.database.rawPassword" -}}
+  {{- if eq .Values.postgresql.enabled true -}}
+    {{- if .Values.postgresql.postgresqlPassword -}}
+      {{- .Values.postgresql.postgresqlPassword -}}
+    {{- else -}}
+      {{ required "A database password is required!" .Values.postgresql.postgresqlPassword }}
```

Comment:
I think it makes sense having a validation function that does all the checks and  prints the proper message

Reply:
➕